cmakelists and windows_buildenv fixes

JosepMaJAZ opened the pull request:
CMakelists.txt: Do not duplicate the runtime binaries on install dir.
windows_buildenv.bat : Fixes on the generation of the CMakeSettings.txt

JosepMaJAZ committed:
CMakelists.txt: Do not duplicate the runtime binaries on install dir.
windows_buildenv.bat : Fixes on the generation of the CMakeSettings.txt

Be-ing commented on CMakeLists.txt:
```diff
@@ -1142,15 +1142,11 @@ if (APPLE)
 else()
+  set(CMAKE_INSTALL_SYSTEM_RUNTIME_DESTINATION "${MIXXX_INSTALL_BINDIR}")
+  if(MSVC AND CMAKE_BUILD_TYPE STREQUAL "Debug")
+    set(CMAKE_INSTALL_DEBUG_LIBRARIES, true)
```
```suggestion
    set(CMAKE_INSTALL_DEBUG_LIBRARIES true)
```

Be-ing commented:
Please fix pre-commit errors.

JosepMaJAZ committed:
Fix comma

Co-authored-by: Be <be.0@gmx.com>

JoergAtGithub commented:
Why you changed the type of CMAKE_PREFIX_PATH from path to string?

JosepMaJAZ commented:
> Why you changed the type of CMAKE_PREFIX_PATH from path to string?

Because PATH means that it is a folder, but here it actually is multiple paths, so it is a string.
Yet, I basically replicated what my manually generated CMakeSettings had.

Anyway, there might be something strange going on there, as the CMakeCache that I generated using cmake has this (UNINITIALIZED)
//No help, variable specified on the command line.
CMAKE_PREFIX_PATH:UNINITIALIZED=D:\programacio\mixxx\mixxx-master\buildenv\2.3-j00019-x64-release-fastbuild-static-55e94982-minimal;D:\programacio\mixxx\mixxx-master\buildenv\2.3-j00019-x64-release-fastbuild-static-55e94982-minimal\Qt-5.14.2

and the one on Visual studio has this (STRING):
CMAKE_PREFIX_PATH:STRING=D:\programacio\mixxx\mixxx-master\buildenv\2.3-j00019-x64-release-fastbuild-static-55e94982-minimal;D:\programacio\mixxx\mixxx-master\buildenv\2.3-j00019-x64-release-fastbuild-static-55e94982-minimal\Qt-5.14.2

JosepMaJAZ committed:
Fixed pre-commit errors and reworked BOM generation.
Also used the BUILD_ROOT and INSTALL_ROOT defined in variables, since those can be outside of the project

JosepMaJAZ committed:
Merge branch 'windows_cmake_fixes' of ssh://github.com/JosepMaJAZ/mixxx into windows_cmake_fixes

JoergAtGithub commented:
Did you know that you can download the patch file with the code style fixes as build artifact? https://github.com/mixxxdj/mixxx/suites/1700507024/artifacts/32000776

JosepMaJAZ committed:
Fixed pre-commit error

Be-ing commented:
I don't know what's going on with this Windows stuff. @JoergAtGithub do you think this is ready to merge?

JoergAtGithub commented:
I do not know the purpose of these console settings. Everything else looks reasonful to me. I did not tested it yet.

JosepMaJAZ commented:
Those console settings are in order to create an UTF8 file with a BOM on it. My Visual Studio created file had such a BOM, so I tried to replicate it as much as I could.

In most cases, it could work without these changes, but this should ensure that it keeps working on other locales.

Holzhaus commented:
https://bugs.launchpad.net/mixxx/+bug/1909873
